Mathlib/GroupTheory/SpecificGroups/Cyclic.lean

```diff
@@ -787,3 +791,105 @@ noncomputable def mulEquivOfPrimeCardEq {p : ℕ} [Fintype G] [Fintype H] [Group
   exact hG.trans hH.symm
 
 end ZMod
+
+section generator
+
+/-!
+### Groups with a given generator
+
+We state some results in terms of an explicitly given generator.
+The generating property is given as in `IsCyclic.exists_generator`.
+
+The main statements are about the existence and uniqueness of homomorphisms and isomorphisms
+specified by the image of the given generator.
+-/
+
+open Subgroup
+
+variable {G G' : Type*} [Group G] [Group G'] {g : G} (hg : ∀ x, x ∈ zpowers g) {g' : G'}
+
+section monoidHom
+
+variable (hg' : orderOf g' ∣ orderOf (g : G))
+
+/-- If `g` generates the group `G` and `g'` is an element of another group `G'` whose order
+divides that of `g`, then there is a homomorphism `G →* G'` mapping `g` to `g'`. -/
+@[to_additive
+   "If `g` generates the additive group `G` and `g'` is an element of another additive group `G'`
+   whose order divides that of `g`, then there is a homomorphism `G →+ G'` mapping `g` to `g'`."]
+noncomputable
+def monoidHom_of_forall_mem_zpowers : G →* G' where
```

Comment:
This should be in snake case as this is a def.

Reply:
Same for the other defs

Reply:
Lower camel case, you mean? 
```text
  4. All other terms of Types (basically anything else) are in lowerCamelCase.
```
(because it is actually snake case). I'll change that.

Reply:
Oh yes. Sorry about that.